app/src/main/java/com/incwelltechnology/lms/ui/auth/AuthViewModel.kt

```diff
@@ -0,0 +1,25 @@
+package com.incwelltechnology.lms.ui.auth
+
+import android.view.View
+import androidx.lifecycle.ViewModel
+import com.incwelltechnology.lms.data.repository.UserRepository
+
+class AuthViewModel : ViewModel() {
+    var username: String? = null
+    var password: String? = null
+
+    var authListener: AuthListener? = null
+
+    fun onLoginButtonClick(view: View) {
```

Comment:
@iamsannyrai do you need view as a parameter?

Reply:
yes or else it will produce databinding errors since it will have problem matching signature ("@{viewmodel::onLoginButtonClick}") which we have provided in onClick() for the login button